src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Trunc.cs

```diff
@@ -79,11 +81,21 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                 if (type0.IsTable)
                 {
                     // Ensure we have a one-column table of numerics
-                    fValid &= CheckNumericColumnType(type0, args[0], errors, ref nodeToCoercedTypeMap);
-
-                    returnType = context.Features.HasFlag(Features.ConsistentOneColumnTableResult)
-                        ? DType.CreateTable(new TypedName(DType.Number, new DName(ColumnName_ValueStr)))
-                        : type0;
+                    fValid &= CheckNumericColumnType(type0, args[0], errors, ref nodeToCoercedTypeMap);
+
+                    if (nodeToCoercedTypeMap?.Any() ?? false)
+                    {
+                        // Now set the coerced type to a table with numeric column type with the same name as in the argument.
+                        returnType = nodeToCoercedTypeMap[args[0]];
+                    }
+                    else
+                    {
+                        returnType = type0;
+                    }
+
+                    returnType = context.Features.HasFlag(Features.ConsistentOneColumnTableResult)
+                        ? DType.CreateTable(new TypedName(DType.Number, new DName(ColumnName_ValueStr)))

```

Comment:
>DType.Number [](http://example.com/codeflow?start=58&length=12)

should these be number (or these function changes also pulled out to the other pr?)

Reply:
I'm going to revert this and skipped the appropriate tests.  We'll make it a part of the other PR [https://github.com/microsoft/Power-Fx/pull/1256](https://github.com/microsoft/Power-Fx/pull/1256)